protocols/ping/src/protocol.rs

```diff
@@ -75,143 +94,126 @@ where
 
     #[inline]
     fn upgrade_outbound(self, socket: Negotiated<TSocket>, _: Self::Info) -> Self::Future {
-        let payload: [u8; 32] = EntropyRng::default().sample(Standard);
-        debug!("Preparing for ping with payload {:?}", payload);
+        let payload = thread_rng().sample(distributions::Standard);
```

Comment:
Nit: I liked the fact that `[u8; 32]` was explicitly written here, as it's an important part of the protocol.

Reply:
Ok, though `[u8; 32]` can be seen in quite a few places in the type signatures in this module already! :)